ballerina/Package.md

```diff
@@ -1,14 +1,466 @@
+`COUNT()` and other SQL aggregation functions). If the provided return type is not a record (i.e., a primitive data type)
+, this operation will return the value of the first column of the first row retrieved by the query.
+
+```ballerina
+int age = 12;
+sql:ParameterizedQuery query = `SELECT COUNT(*) FROM students WHERE age < ${age}`;
+int youngStudents = check dbClient->queryRow(query);
+```
+
+#### Updating Data
+
+This sample demonstrates modifying data by executing an `UPDATE` statement via the `execute` remote function of
+the client.
+
+```ballerina
+int age = 23;
+sql:ParameterizedQuery query = `UPDATE students SET name = 'John' WHERE age = ${age}`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+#### Deleting Data
+
+This sample demonstrates deleting data by executing a `DELETE` statement via the `execute` remote function of
+the client.
+
+```ballerina
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This sample demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and
+parameterized SQL query as same as the above `execute` operations.
+
+```ballerina
+// Create the table with the records that need to be inserted.
+var data = [
+  { name: "John", age: 25 },
+  { name: "Peter", age: 24 },
+  { name: "jane", age: 22 }
+];
+
+// Do the batch update by passing the batches.
+sql:ParameterizedQuery[] batch = from var row in data
+                                 select `INSERT INTO students ('name', 'age')
+                                           VALUES (${row.name}, ${row.age})`;
+sql:ExecutionResult[] result = check dbClient->batchExecute(batch);
+```
+
+#### Execute SQL Stored Procedures
+
+This sample demonstrates how to execute a stored procedure with a single `INSERT` statement that is executed via the
+`call` remote function of the client.
+
+```ballerina
+int uid = 10;
+sql:IntegerOutParameter insertId = new;
+
+sql:ProcedureCallResult result = 
+                         check dbClient->call(`call InsertPerson(${uid}, ${insertId})`);
+stream<record{}, sql:Error?>? resultStr = result.queryResult;
+if resultStr is stream<record{}, sql:Error?> {
+   check from record{} result in resultStr
+      do {
+         // Can perform operations using the record 'result'.
+      };
+}
+check result.close();
+```
+Note that you have to invoke the close operation explicitly on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
+
+>**Note:** The default thread pool size used in Ballerina is: `the number of processors available * 2`. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
 
 ### Report Issues
```

Comment:
```suggestion
## Report Issues
```

Reply:
Fixed in [b02d390](https://github.com/ballerina-platform/module-ballerinax-mssql/pull/163/commits/b02d39058c4c5fdd3204febac47ec271b869d81b)